pytest_localftpserver/servers.py

```diff
@@ -77,35 +76,41 @@ def __init__(
             self._ftp_home = ftp_home
         self.username = username
         self.password = password
+        self._uses_TLS = use_TLS
+        self._cert_path = certfile
+        self._socket, self._ftp_port = get_socket(ftp_port)
+        self.shutdown_event = multiprocessing.Event()
+
+    def run(self):
         authorizer = DummyAuthorizer()
         authorizer.add_user(
             self.username, self.password, self._ftp_home, perm="elradfmwM"
         )
         authorizer.add_anonymous(self._anon_root)
 
-        self._uses_TLS = use_TLS
-        self._cert_path = certfile
-
-        if use_TLS:
+        if self._uses_TLS:
             handler = TLS_FTPHandler
-            handler.certfile = certfile
-            validate_cert_file(certfile)
+            handler.certfile = self._cert_path
+            validate_cert_file(self._cert_path)
         else:
             handler = FTPHandler
-
-        socket, self._ftp_port = get_socket(ftp_port)
-
         handler.authorizer = authorizer
 
         # Create a new pyftpdlib server with the socket and handler we've
         # configured
-        FTPServer.__init__(self, socket, handler)
+        FTPServer.__init__(self, self._socket, handler)
```

Comment:
The key to solving the problem is to defer the initialization of `FTPServer` to the child process, so that the ioloop starts in the child process instead of in the parent process.